### DISCUSSION TITLE
Container element not found

### DISCUSSION THREAD

**Discussion by NinaP48** [katspaugh/wavesurfer.js#2719] [Q&A] [answered]
Bonjour, j'essaie de visualiser une forme d'onde sonore à l'aide de wavesurfer. Cependant, rien n'est chargé en javascript sur ma page. Je n'arrive pas à résoudre ce problème.

Voici le code HTML : 
> <!DOCTYPE html>
<html lang="FR-fr">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">

    <script src="https://unpkg.com/wavesurfer.js@4.2.0/dist/wavesurfer.js"></script>
	  <script src="index.js"></script>
  </head>
  <body>
    <main>
        <h1>WaveForm</h1>  
    </main>
    

    <div id ="waveform"></div>
  </body>
</html>

Et voici le code Javascript : 

var wavesurfer = WaveSurfer.create({
    container: "#waveform",
    waveColor: 'violet',
    progressColor: 'purple',
    scrollParent: true
});

wavesurfer.load('ReveilDesOiseaux.wav');

wavesurfer.on('ready', function () {
    wavesurfer.play();
});

**Comment by katspaugh** [comment]
Make sure your script is running after the document has loaded.
A common way to ensure this is to put the script to the bottom of your HTML.
Alternatively, you can wrap the wavesurfer init code into this:
```
window.addEventListener('DOMContentLoaded', () => {
  var wavesurfer = WaveSurfer.create({ ... })
})
```

**Comment by NinaP48** [comment]
J'ai effectué les modifications que vous me conseillez cependant, cela me génère une autre erreur : 
![Capture d’écran 2023-03-13 114051](https://user-images.githubusercontent.com/114979301/224679151-ba04a12e-0ed8-44fb-904c-772bcc957452.png)


Voici le code HTML :

<!DOCTYPE html>
<html lang="FR-fr">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Maquette WaveForm</title>

  </head>
  <body>
    <main>
        <h1>WaveForm</h1>  
    </main>
    

    <div id ="waveform"></div>
  </body>
</html>
<script src="https://unpkg.com/wavesurfer.js@4.2.0/dist/wavesurfer.js"></script>

<script src="index.js"></script>


Voici le code javascript :


window.addEventListener('DOMContentLoaded', () => {
    var wavesurfer = WaveSurfer.create({
        container: "#waveform",
        waveColor: 'violet',
        progressColor: 'purple',
        scrollParent: true
    })
});

wavesurfer.load('ReveilDesOiseaux.wav');


wavesurfer.on('ready', function () {
    wavesurfer.play();
});



Je m'excuse si je n'ai pas compris ce que vous m'avez indiqué, je débute totalement en javascript.

**Comment by katspaugh** [comment, marked as answer]
Das Problem ist, dass das `wavesurfer.load` auch im `DOMContentLoaded` sein muss, also:

```
window.addEventListener('DOMContentLoaded', () => {
  var wavesurfer = WaveSurfer.create({ ... })
  wavesurfer.load('ReveilDesOiseaux.wav');
})
```

Ich hoffe, dass es dir hilft! Viel Glück mit JavaScript.

**Reply by NinaP48** [reply]
Super, merci beaucoup 
Maintenant je dois encore résoudre les problèmes avec les CORS ! :)

**Reply by katspaugh** [reply]
No problem!
Yeah, CORS won't work with file://. You need to run an HTTP server locally, or upload your audio file somewhere.

Btw, you seem to be using a rather old wavesurfer.js version. This will load the latest version (6.5.2):
```
<script src="https://unpkg.com/wavesurfer.js"></script>
```
